### DISCUSSION TITLE
[PR] Adding auto RTL support

### DISCUSSION THREAD

**Discussion by FaridAghili** [laravel/nova-issues#4306] [Ideas & Feature Requests]
Here I'm going to describe an easy and automated way for adding RTL theme to Nova v4.

**A little background:** I've developed a theme for Nova v3 to add RTL theme 3 years ago and it still works with latest v3 branch, it has ~23k installs:
https://github.com/pishran/nova-rtl-theme

For Nova 4 support, I've tried to find a better approach and here is my suggestion:

1. Add [postcss-rtlcss](https://github.com/elchininet/postcss-rtlcss) package to Nova's `package.json`.
2.  Add `postcssRtlcss` to Nova's `webpack.mix.js` as a postCss plugin:
```js
let postcssRtlcss = require('postcss-rtlcss')
...
.postCss('resources/css/app.css', 'public', [postcssImport(), tailwindcss('tailwind.config.js'), postcssRtlcss()])
```
3. Add a new key to Nova's config file for toggling RTL theme:
```php
...
'rtl' => true, // Defaults to false
...
```
4. In Nova's `layout.blade.php` file, add a `dir` attribute to `html` tag according to the config key added in step 3:
```blade
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}" class="h-full font-sans antialiased"
      dir="{{ config('nova.rtl') ? 'rtl' : 'ltr' }}">
...
```

Done!

In this approach, no manual tweaking is required for users and it's also updates with Nova's internal developments when css codes change.

Let me know if there is anything I can do.
@taylorotwell @davidhemphill @crynobone

**Comment by FaridAghili** [comment]
Some screenshots:

![image](https://user-images.githubusercontent.com/8607021/170153422-d2f47399-8405-45fc-bcb1-2f43daf684a9.png)

![image](https://user-images.githubusercontent.com/8607021/170153448-9a287271-a9de-43c9-b1ba-7fb3c120e673.png)

![image](https://user-images.githubusercontent.com/8607021/170153465-54498151-3c76-4ed2-9655-f8559f5d4887.png)

![image](https://user-images.githubusercontent.com/8607021/170153507-daef164c-6bc1-45c7-92fa-1cc7ffdb7f89.png)

![image](https://user-images.githubusercontent.com/8607021/170153524-a2d401a4-578b-4d81-b21b-ab2320466c52.png)

**Comment by mustafa-online** [comment]
This is looking great 😍🌟

**Comment by sinamiandashti** [comment]
awsome - +1 for this

**Comment by emadbatwa** [comment]
great work 👍 
+1000

**Comment by sajaddp** [comment]
Excellent😍

**Comment by sinamiandashti** [comment]
any update on this?

**Comment by crynobone** [comment]
We currently working on this.

**Comment by sajaddp** [comment]
Update 4.10.0: Added preliminary support for right-to-left languages with `Nova::enableRTL()`.

@crynobone Thank you.
